ftva_lab_data/views.py

```diff
@@ -524,3 +524,51 @@ def get_external_search_results(
             "Invalid search type specified.",
             status=400,
         )
+
+
+def generate_metadata_json(
+    request: HttpRequest, record_id: int, inventory_number: str
+) -> JsonResponse | str:
+    """Generate a  JSON metadata record for a given inventory number,
+    by combining data from Alma, Filemaker, and Django.
+
+    :param request: The HTTP request object.
+    :param record_id: The ID of the Django record to use.
+    :param inventory_number: The inventory number to search for in Alma and Filemaker.
+    :return: A JSON record containing the combined metadata.
+    """
+
+    # Get Django record
```

Comment:
The logic to get the record here (the dict required by the metadata function) should be the same as the logic to get it via the REST-ish API call used by outside callers.
This logic currently is most of the code in `views.get_record()`, so time to pull that into a separate method in `views_utils`.  That way, Django data passed to the metadata function for a given record will be identical, regardless of caller.